src/pages/workspace/receiptPartners/InviteReceiptPartnerPolicyPage.tsx

```diff
@@ -0,0 +1,250 @@
+            if (personalDetail) {
+                const memberForList = formatMemberForList({
+                    text: personalDetail?.displayName ?? email,
+                    alternateText: email,
+                    login: email,
+                    accountID: personalDetail?.accountID,
+                    icons: [
+                        {
+                            source: personalDetail?.avatar ?? Expensicons.FallbackAvatar,
+                            name: formatPhoneNumber(email),
+                            type: CONST.ICON_TYPE_AVATAR,
+                            id: personalDetail?.accountID,
+                        },
+                    ],
+                    reportID: '',
+                    keyForList: email,
+                });
+
+                membersList.push({
+                    ...memberForList,
+                    isSelected: true,
+                });
+            }
+        });
+
+        membersList = sortAlphabetically(membersList, 'text', localeCompare);
+
+        return membersList;
+    }, [isOffline, policy?.employeeList, localeCompare]);
+
+    const sections = useMemo(() => {
+        if (workspaceMembers.length === 0) {
+            return [];
+        }
+
+        let membersToDisplay = workspaceMembers;
+
+        // Apply search filter if there's a search term
+        if (debouncedSearchTerm) {
+            const searchValue = getSearchValueForPhoneOrEmail(debouncedSearchTerm).toLowerCase();
+            membersToDisplay = tokenizedSearch(workspaceMembers, searchValue, (option) => [option.text ?? '', option.alternateText ?? '']);
+        }
+
+        // Filter to show selected members first, then apply search filter to selected members
+        let filterSelectedOptions = selectedOptions;
+        if (debouncedSearchTerm !== '') {
+            const searchValue = getSearchValueForPhoneOrEmail(debouncedSearchTerm).toLowerCase();
+            filterSelectedOptions = selectedOptions.filter((option) => {
+                const isPartOfSearchTerm = !!option.text?.toLowerCase().includes(searchValue) || !!option.login?.toLowerCase().includes(searchValue);
+                return isPartOfSearchTerm;
+            });
+        }
+
+        // Combine selected members with unselected members
+        const selectedLogins = selectedOptions.map(({login}) => login);
+        const unselectedMembers = membersToDisplay.filter(({login}) => !selectedLogins.includes(login));
+
+        // Update members with current selection state
+        const selectedMembersWithState = filterSelectedOptions.map((member) => ({
+            ...member,
+            isSelected: true,
+        }));
+
+        const unselectedMembersWithState = unselectedMembers.map((member) => ({
+            ...member,
+            isSelected: false,
+        }));
+
+        // Combine all members (selected first, then unselected)
+        const allMembersWithState = [...selectedMembersWithState, ...unselectedMembersWithState];
```

Comment:
The intermediate variables are not needed and this avoids copying the full objects again:
```suggestion
		const allMembersWithState = [];
        filterSelectedOptions.forEach((member) => {
            allMembersWithState.push(
                ...member,
                isSelected: true,
            );
        });
        unselectedMembers.forEach((member) => {
            allMembersWithState.push(
                ...member,
                isSelected: false,
            );
        });
```